File: TP1/pixel_java/Image.java
public class Image {
    private int width;
    private int height;
    private Pixel[][] tab;

    public Image() {
        this(0, 0);
    }

    public Image(int width, int height) {
        this.width = width;
        this.height = height;
        this.tab = new Pixel[height][width];
        for (int i = 0; i < height; i++) {
            for (int j = 0; j < width; j++) {
                this.tab[i][j] = new Pixel();
            }
        }
    }

    public int getWidth() {
        return this.width;
    }

    public int getHeight() {
        return this.height;
    }

    public void setWidth(int width) {
        this.width = width;
    }

    public void setHeight(int height) {
        this.height = height;
    }

    public Pixel getPixel(int i, int j) {
        return this.tab[i][j];
    }

    public void setPixel(int i, int j, Pixel p) {
        this.tab[i][j] = p;
    }
    
}

File: TP1/pixel_cpp/image.cpp
#include "image.h"
#include <iostream>

Image::Image() : Image(0, 0) {}
Image::Image(const unsigned int& w, const unsigned int& h) : width(w), height(h) {
    tab = new Pixel*[width];
    for (unsigned int i = 0; i < width; i++) {
        tab[i] = new Pixel[height];
    }

}
Image::~Image() {
    if (tab != nullptr) {
        for (unsigned int i = 0; i < width; i++) {
            delete tab[i];
        }
    delete [] tab;
    }
    else std::cout << "bip" << std::endl;
}

unsigned int Image::getWidth() const {
    return width;
}

unsigned int Image::getHeight() const {
    return height;
}

Pixel Image::getPixel(const int& x, const int& y) const {
    return tab[x][y];
}

void Image::setWidth(const unsigned int& w) {
    width = w;
}

void Image::setHeight(const unsigned int& h) {
    height = h;
}

void Image::setPixel(const int& x, const int& y, const Pixel& p) {
    tab[x][y] = p;
}